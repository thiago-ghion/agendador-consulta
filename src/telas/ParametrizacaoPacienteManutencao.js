import React, { useState, useRef, useEffect } from "react";
import { Button } from "react-bootstrap";
import Row from "react-bootstrap/Row";
import Col from "react-bootstrap/Col";
import Form from "react-bootstrap/Form";
import InputMask from "react-input-mask";
import { useDispatch, useSelector } from "react-redux";

import {
  registrarPacienteAction,
  consultarPacienteAction,
  alterarPacienteAction,
} from "../features/pacienteSlice";

function ParametrizacaoPacienteManutencao(props) {
  const dispatch = useDispatch();

  const { paciente } = useSelector((state) => state.paciente);

  const [nomePaciente, setNomePaciente] = useState("");
  const [cpf, setCPF] = useState("");
  const [dataNascimento, setDataNascimento] = useState("");
  const [telefone, setTelefone] = useState("");
  const [email, setEmail] = useState("");

  const telaCarregada = useRef(false);

  useEffect(() => {
    if (telaCarregada.current === false) {
      if (props.modoAlteracao) {
        const acao = async () => {
          const resposta = await dispatch(
            consultarPacienteAction({ idPaciente: props.idPaciente })
          );
          if (resposta.error === undefined) {
            setNomePaciente(resposta.payload.nomePaciente);
            setCPF(resposta.payload.numeroCPF);
            setDataNascimento(resposta.payload.dataNascimento);
            if (resposta.payload.numeroTelefone.length === 10) {
              setTelefone(
                `${resposta.payload.numeroTelefone.substring(
                  0,
                  2
                )} ${resposta.payload.numeroTelefone.substring(2, 10)}`
              );
            } else {
              setTelefone(resposta.payload.numeroTelefone);
            }
            setEmail(resposta.payload.enderecoEmail);
          }
        };
        acao();
      }
    }
    telaCarregada.current = true;
  });

  let nomePacienteInput;
  let cpfInput;
  let dataNascimentoInput;
  let telefoneInput;
  let emailInput;

  const dataPattern = [
    /[0-3]/,
    /[0-9]/,
    "/",
    /[0-1]/,
    /[0-9]/,
    "/",
    /[1-2]/,
    /[0-9]/,
    /[0-9]/,
    /[0-9]/,
  ];

  const telefonePattern = [
    "(",
    /[1-9]/,
    /[1-9]/,
    ")",
    /[9,\s]/,
    /[0-9]/,
    /[0-9]/,
    /[0-9]/,
    /[0-9]/,
    "-",
    /[0-9]/,
    /[0-9]/,
    /[0-9]/,
    /[0-9]/,
  ];

  const formatarCampo = (texto, isNumerico) => {
    const valor = texto.replaceAll(/[.\-_\(\)\s]/g, "");
    if (valor === "") {
      return undefined;
    }
    if (isNumerico) {
      return +valor;
    }
    return valor;
  };

  const salvar = async () => {
    const requisicao = {
      nomePaciente: nomePaciente,
      numeroCPF: formatarCampo(cpf, true),
      dataNascimento: dataNascimento.replaceAll(/[\/]/g, "."),
      numeroTelefone: formatarCampo(telefone, false),
      enderecoEmail: email,
    };
    let resposta;

    if (props.modoAlteracao) {
      resposta = await dispatch(
        alterarPacienteAction({ idPaciente: props.idPaciente, requisicao })
      );
    } else {
      resposta = await dispatch(registrarPacienteAction({ requisicao }));
    }

    if (resposta.error === undefined) {
      props.setTelaAtiva(1);
    } else {
      if (resposta.payload.campo !== undefined) {
        switch (resposta.payload.campo) {
          case 1:
            nomePacienteInput.focus();
            break;
          case 2:
            cpfInput.focus();
            break;
          case 3:
            dataNascimentoInput.focus();
            break;
          case 4:
            telefoneInput.focus();
            break;
          case 5:
            emailInput.focus();
            break;
          default:
            break;
        }
      }
    }
  };

  return (
    <div>
      <br></br>
      <Row>
        <Col>
          <Form.Group className="mb-3">
            <Form.Control
              type="text"
              placeholder="Informe o nome do paciente"
              value={nomePaciente}
              autoFocus
              maxLength="150"
              onChange={(event) => {
                setNomePaciente(event.target.value);
              }}
              ref={(input) => {
                if (input !== null) {
                  nomePacienteInput = input;
                }
              }}
            />
          </Form.Group>
        </Col>
      </Row>
      <Row>
        <Col md={4}>
          <Form.Group className="mb-3">
            <Form.Control
              placeholder="Informe o CPF do paciente"
              value={cpf}
              as={InputMask}
              mask="999.999.999-99"
              onChange={(event) => {
                setCPF(event.target.value);
              }}
              ref={(input) => {
                if (input !== null) {
                  cpfInput = input;
                }
              }}
            />
          </Form.Group>
        </Col>
        <Col md={4}>
          <Form.Group className="mb-3">
            <Form.Control
              type="text"
              as={InputMask}
              mask={dataPattern}
              placeholder="Informe a data de nascimento"
              value={dataNascimento}
              onChange={(event) => {
                setDataNascimento(event.target.value);
              }}
              ref={(input) => {
                if (input !== null) {
                  dataNascimentoInput = input;
                }
              }}
            />
          </Form.Group>
        </Col>
        <Col md={4}>
          <Form.Group className="mb-3">
            <Form.Control
              type="text"
              placeholder="Informe a telefone"
              value={telefone}
              as={InputMask}
              mask={telefonePattern}
              onChange={(event) => {
                setTelefone(event.target.value);
              }}
              ref={(input) => {
                if (input !== null) {
                  telefoneInput = input;
                }
              }}
            />
          </Form.Group>
        </Col>
      </Row>
      <Row>
        <Col>
          <Form.Group className="mb-3">
            <Form.Control
              type="text"
              placeholder="Informe a email"
              value={email}
              maxLength="150"
              onChange={(event) => {
                setEmail(event.target.value);
              }}
              ref={(input) => {
                if (input !== null) {
                  emailInput = input;
                }
              }}
            />
          </Form.Group>
        </Col>
      </Row>
      <Row>
        <Col md={1}>
          <Button variant="success" onClick={salvar}>
            Salvar
          </Button>
        </Col>
        <Col md={1}>
          <Button
            variant="danger"
            onClick={() => {
              props.setTelaAtiva(1);
            }}
          >
            Cancelar
          </Button>
        </Col>
      </Row>
    </div>
  );
}

export default ParametrizacaoPacienteManutencao;
