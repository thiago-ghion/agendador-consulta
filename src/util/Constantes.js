const URL_API = 'http://localhost:5000';

export { URL_API };