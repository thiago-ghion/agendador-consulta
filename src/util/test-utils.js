import React from "react";
import { render } from "@testing-library/react";
import { configureStore } from "@reduxjs/toolkit";
import { Provider } from "react-redux";
// As a basic setup, import your same slice reducers
import loginReducer from "../features/loginSlice";
import profissionalReducer from "../features/profissionalSlice";
import consultaReducer from "../features/consultaSlice";
import pacienteReducer from "../features/pacienteSlice";
import horarioReducer from "../features/horarioSlice";
import usuarioReducer from "../features/usuarioSlice";
import loadingReducer from "../features/loadingSlice";
import mensagemgReducer from "../features/mensagemSlice";

export function renderWithProviders(
  ui,
  {
    preloadedState = {},
    // Automatically create a store instance if no store was passed in
    store = configureStore({
      reducer: {
        login: loginReducer,
        consulta: consultaReducer,
        profissional: profissionalReducer,
        paciente: pacienteReducer,
        horario: horarioReducer,
        usuario: usuarioReducer,
        loading: loadingReducer,
        mensagem: mensagemgReducer,
      },
      preloadedState,
    }),
    ...renderOptions
  } = {}
) {
  function Wrapper({ children }) {
    return <Provider store={store}>{children}</Provider>;
  }

  // Return an object with the store and all of RTL's query functions
  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
}
