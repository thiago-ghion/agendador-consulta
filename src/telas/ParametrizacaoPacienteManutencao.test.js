import React from "react";
import { rest } from "msw";
import { setupServer } from "msw/node";
import { renderWithProviders } from "../util/test-utils";

import ParametrizacaoPacienteManutencao from "./ParametrizacaoPacienteManutencao";
import { screen, fireEvent, waitFor } from "@testing-library/react";

let mockConsultar = {};
let estadoConsultar = 200;
let mockRegistrar = {};
let estadoRegistrar = 201;
let mockAlterar = {};
let estadoAlterar = 200;

export const handlers = [
  rest.get("http://localhost:5000/v1/paciente/consultar", (req, res, ctx) => {
    return res(ctx.status(estadoConsultar), ctx.json(mockConsultar));
  }),
  rest.get("http://localhost:5000/v1/paciente/consultar", (req, res, ctx) => {
    return res(ctx.status(estadoConsultar), ctx.json(mockConsultar));
  }),
  rest.post("http://localhost:5000/v1/paciente/registrar", (req, res, ctx) => {
    return res(ctx.status(estadoRegistrar), ctx.json(mockRegistrar));
  }),
  rest.post("http://localhost:5000/v1/paciente/alterar/1", (req, res, ctx) => {
    return res(ctx.status(estadoAlterar), ctx.json(mockAlterar));
  }),
];

const server = setupServer(...handlers);

// Enable API mocking before tests.
beforeAll(() => server.listen());

// Reset any runtime request handlers we may add during the tests.
afterEach(() => server.resetHandlers());

// Disable API mocking after the tests are done.
afterAll(() => server.close());

beforeEach(() => {
  mockConsultar = {};
  estadoConsultar = 200;
  mockRegistrar = {};
  estadoRegistrar = 201;
  mockAlterar = {};
  estadoAlterar = 200;
});

test("Renderização componente", async () => {
  expect(
    renderWithProviders(<ParametrizacaoPacienteManutencao />)
  ).toMatchSnapshot();
});

test("Renderizar em modo de alteração - Telefone móvel", async () => {
  mockConsultar = {
    nomePaciente: "Teste",
    numeroCPF: "12345678901",
    dataNascimento: "01.01.2020",
    numeroTelefone: "11912345678",
    enderecoEmail: "teste@teste.com",
  };
  renderWithProviders(
    <ParametrizacaoPacienteManutencao
      modoAlteracao
      idPaciente={1}
      setTelaAtiva={() => {}}
    />
  );
});

test("Renderizar em modo de alteração - Telefone fixo", async () => {
  mockConsultar = {
    nomePaciente: "Teste",
    numeroCPF: "12345678901",
    dataNascimento: "01.01.2020",
    numeroTelefone: "1112345678",
    enderecoEmail: "teste@teste.com",
  };
  renderWithProviders(
    <ParametrizacaoPacienteManutencao
      modoAlteracao
      idPaciente={1}
      setTelaAtiva={() => {}}
    />
  );
});

test("Renderizar em modo de alteração - Erro", async () => {
  estadoConsultar = 400;
  mockConsultar = {};
  renderWithProviders(
    <ParametrizacaoPacienteManutencao
      modoAlteracao
      idPaciente={1}
      setTelaAtiva={() => {}}
    />
  );
});

test("Cancelar manutenção", async () => {
  renderWithProviders(
    <ParametrizacaoPacienteManutencao setTelaAtiva={() => {}} />
  );
  const cancelar = await waitFor(() =>
    screen.findByRole("button", { name: /Cancelar/i })
  );
  fireEvent.click(cancelar);
});

test("Preencher o formulário", async () => {
  renderWithProviders(
    <ParametrizacaoPacienteManutencao setTelaAtiva={() => {}} />
  );

  fireEvent.change(screen.getByPlaceholderText("Informe o nome do paciente"), {
    target: { value: "teste" },
  });
  fireEvent.change(screen.getByPlaceholderText("Informe o CPF do paciente"), {
    target: { value: "12345678901" },
  });
  fireEvent.change(
    screen.getByPlaceholderText("Informe a data de nascimento"),
    { target: { value: "01012020" } }
  );
  fireEvent.change(screen.getByPlaceholderText("Informe a telefone"), {
    target: { value: "11912345678" },
  });
  fireEvent.change(screen.getByPlaceholderText("Informe a email"), {
    target: { value: "teste@teste.com" },
  });
});

test("Inclusão do paciente com sucesso", async () => {
  renderWithProviders(
    <ParametrizacaoPacienteManutencao setTelaAtiva={() => {}} />
  );

  fireEvent.change(screen.getByPlaceholderText("Informe o nome do paciente"), {
    target: { value: "teste" },
  });
  fireEvent.change(screen.getByPlaceholderText("Informe o CPF do paciente"), {
    target: { value: "123.456.789-01" },
  });
  fireEvent.change(
    screen.getByPlaceholderText("Informe a data de nascimento"),
    { target: { value: "01/01/2020" } }
  );
  fireEvent.change(screen.getByPlaceholderText("Informe a telefone"), {
    target: { value: "(11)91234-5678" },
  });
  fireEvent.change(screen.getByPlaceholderText("Informe a email"), {
    target: { value: "teste@teste.com" },
  });

  const botaoSalvar = await waitFor(() =>
    screen.findByRole("button", { name: /Salvar/i })
  );
  fireEvent.click(botaoSalvar);
});

test("Alteração do paciente com sucesso", async () => {
  renderWithProviders(
    <ParametrizacaoPacienteManutencao
      idPaciente={1}
      modoAlteracao
      setTelaAtiva={() => {}}
    />
  );

  fireEvent.change(screen.getByPlaceholderText("Informe o nome do paciente"), {
    target: { value: "teste" },
  });
  fireEvent.change(screen.getByPlaceholderText("Informe o CPF do paciente"), {
    target: { value: "123.456.789-01" },
  });
  fireEvent.change(
    screen.getByPlaceholderText("Informe a data de nascimento"),
    { target: { value: "01/01/2020" } }
  );
  fireEvent.change(screen.getByPlaceholderText("Informe a telefone"), {
    target: { value: "(11)91234-5678" },
  });
  fireEvent.change(screen.getByPlaceholderText("Informe a email"), {
    target: { value: "teste@teste.com" },
  });

  const botaoSalvar = await waitFor(() =>
    screen.findByRole("button", { name: /Salvar/i })
  );
  fireEvent.click(botaoSalvar);
});

test("Manutenção do paciente erro no formulário", async () => {
  renderWithProviders(
    <ParametrizacaoPacienteManutencao
      modoAlteracao
      idPaciente={1}
      setTelaAtiva={() => {}}
    />
  );

  fireEvent.change(screen.getByPlaceholderText("Informe o nome do paciente"), {
    target: { value: "teste" },
  });
  fireEvent.change(screen.getByPlaceholderText("Informe o CPF do paciente"), {
    target: { value: "" },
  });
  fireEvent.change(
    screen.getByPlaceholderText("Informe a data de nascimento"),
    { target: { value: "01/01/2020" } }
  );
  fireEvent.change(screen.getByPlaceholderText("Informe a telefone"), {
    target: { value: "(11)91234-5678" },
  });
  fireEvent.change(screen.getByPlaceholderText("Informe a email"), {
    target: { value: "teste@teste.com" },
  });

  const botaoSalvar = await waitFor(() =>
    screen.findByRole("button", { name: /Salvar/i })
  );
  fireEvent.click(botaoSalvar);
});

test("Manutenção do paciente - erro - nome", async () => {
  estadoRegistrar = 400;
  mockRegistrar = { campo: 1 };

  renderWithProviders(
    <ParametrizacaoPacienteManutencao setTelaAtiva={() => {}} />
  );

  fireEvent.change(screen.getByPlaceholderText("Informe o nome do paciente"), {
    target: { value: "teste" },
  });
  fireEvent.change(screen.getByPlaceholderText("Informe o CPF do paciente"), {
    target: { value: "123.456.789-01" },
  });
  fireEvent.change(
    screen.getByPlaceholderText("Informe a data de nascimento"),
    { target: { value: "01/01/2020" } }
  );
  fireEvent.change(screen.getByPlaceholderText("Informe a telefone"), {
    target: { value: "(11)91234-5678" },
  });
  fireEvent.change(screen.getByPlaceholderText("Informe a email"), {
    target: { value: "teste@teste.com" },
  });

  const botaoSalvar = await waitFor(() =>
    screen.findByRole("button", { name: /Salvar/i })
  );
  fireEvent.click(botaoSalvar);
});

test("Manutenção do paciente - erro - cpf", async () => {
  estadoRegistrar = 400;
  mockRegistrar = { campo: 2 };

  renderWithProviders(
    <ParametrizacaoPacienteManutencao setTelaAtiva={() => {}} />
  );

  fireEvent.change(screen.getByPlaceholderText("Informe o nome do paciente"), {
    target: { value: "teste" },
  });
  fireEvent.change(screen.getByPlaceholderText("Informe o CPF do paciente"), {
    target: { value: "123.456.789-01" },
  });
  fireEvent.change(
    screen.getByPlaceholderText("Informe a data de nascimento"),
    { target: { value: "01/01/2020" } }
  );
  fireEvent.change(screen.getByPlaceholderText("Informe a telefone"), {
    target: { value: "(11)91234-5678" },
  });
  fireEvent.change(screen.getByPlaceholderText("Informe a email"), {
    target: { value: "teste@teste.com" },
  });

  const botaoSalvar = await waitFor(() =>
    screen.findByRole("button", { name: /Salvar/i })
  );
  fireEvent.click(botaoSalvar);
});

test("Manutenção do paciente - erro - data nascimento", async () => {
  estadoRegistrar = 400;
  mockRegistrar = { campo: 3 };

  renderWithProviders(
    <ParametrizacaoPacienteManutencao setTelaAtiva={() => {}} />
  );

  fireEvent.change(screen.getByPlaceholderText("Informe o nome do paciente"), {
    target: { value: "teste" },
  });
  fireEvent.change(screen.getByPlaceholderText("Informe o CPF do paciente"), {
    target: { value: "123.456.789-01" },
  });
  fireEvent.change(
    screen.getByPlaceholderText("Informe a data de nascimento"),
    { target: { value: "01/01/2020" } }
  );
  fireEvent.change(screen.getByPlaceholderText("Informe a telefone"), {
    target: { value: "(11)91234-5678" },
  });
  fireEvent.change(screen.getByPlaceholderText("Informe a email"), {
    target: { value: "teste@teste.com" },
  });

  const botaoSalvar = await waitFor(() =>
    screen.findByRole("button", { name: /Salvar/i })
  );
  fireEvent.click(botaoSalvar);
});

test("Manutenção do paciente - erro - telefone", async () => {
  estadoRegistrar = 400;
  mockRegistrar = { campo: 4 };

  renderWithProviders(
    <ParametrizacaoPacienteManutencao setTelaAtiva={() => {}} />
  );

  fireEvent.change(screen.getByPlaceholderText("Informe o nome do paciente"), {
    target: { value: "teste" },
  });
  fireEvent.change(screen.getByPlaceholderText("Informe o CPF do paciente"), {
    target: { value: "123.456.789-01" },
  });
  fireEvent.change(
    screen.getByPlaceholderText("Informe a data de nascimento"),
    { target: { value: "01/01/2020" } }
  );
  fireEvent.change(screen.getByPlaceholderText("Informe a telefone"), {
    target: { value: "(11)91234-5678" },
  });
  fireEvent.change(screen.getByPlaceholderText("Informe a email"), {
    target: { value: "teste@teste.com" },
  });

  const botaoSalvar = await waitFor(() =>
    screen.findByRole("button", { name: /Salvar/i })
  );
  fireEvent.click(botaoSalvar);
});

test("Manutenção do paciente - erro - email", async () => {
  estadoRegistrar = 400;
  mockRegistrar = { campo: 5 };

  renderWithProviders(
    <ParametrizacaoPacienteManutencao setTelaAtiva={() => {}} />
  );

  fireEvent.change(screen.getByPlaceholderText("Informe o nome do paciente"), {
    target: { value: "teste" },
  });
  fireEvent.change(screen.getByPlaceholderText("Informe o CPF do paciente"), {
    target: { value: "123.456.789-01" },
  });
  fireEvent.change(
    screen.getByPlaceholderText("Informe a data de nascimento"),
    { target: { value: "01/01/2020" } }
  );
  fireEvent.change(screen.getByPlaceholderText("Informe a telefone"), {
    target: { value: "(11)91234-5678" },
  });
  fireEvent.change(screen.getByPlaceholderText("Informe a email"), {
    target: { value: "teste@teste.com" },
  });

  const botaoSalvar = await waitFor(() =>
    screen.findByRole("button", { name: /Salvar/i })
  );
  fireEvent.click(botaoSalvar);
});
