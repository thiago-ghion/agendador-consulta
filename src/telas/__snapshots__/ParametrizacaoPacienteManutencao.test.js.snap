// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renderização componente 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <br />
        <div
          class="row"
        >
          <div
            class="col"
          >
            <div
              class="mb-3"
            >
              <input
                class="form-control"
                maxlength="150"
                placeholder="Informe o nome do paciente"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
        <div
          class="row"
        >
          <div
            class="col-md-4"
          >
            <div
              class="mb-3"
            >
              <input
                class="form-control"
                placeholder="Informe o CPF do paciente"
                value=""
              />
            </div>
          </div>
          <div
            class="col-md-4"
          >
            <div
              class="mb-3"
            >
              <input
                class="form-control"
                placeholder="Informe a data de nascimento"
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class="col-md-4"
          >
            <div
              class="mb-3"
            >
              <input
                class="form-control"
                placeholder="Informe a telefone"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
        <div
          class="row"
        >
          <div
            class="col"
          >
            <div
              class="mb-3"
            >
              <input
                class="form-control"
                maxlength="150"
                placeholder="Informe a email"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
        <div
          class="row"
        >
          <div
            class="col-md-1"
          >
            <button
              class="btn btn-success"
              type="button"
            >
              Salvar
            </button>
          </div>
          <div
            class="col-md-1"
          >
            <button
              class="btn btn-danger"
              type="button"
            >
              Cancelar
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <br />
      <div
        class="row"
      >
        <div
          class="col"
        >
          <div
            class="mb-3"
          >
            <input
              class="form-control"
              maxlength="150"
              placeholder="Informe o nome do paciente"
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="col-md-4"
        >
          <div
            class="mb-3"
          >
            <input
              class="form-control"
              placeholder="Informe o CPF do paciente"
              value=""
            />
          </div>
        </div>
        <div
          class="col-md-4"
        >
          <div
            class="mb-3"
          >
            <input
              class="form-control"
              placeholder="Informe a data de nascimento"
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          class="col-md-4"
        >
          <div
            class="mb-3"
          >
            <input
              class="form-control"
              placeholder="Informe a telefone"
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="col"
        >
          <div
            class="mb-3"
          >
            <input
              class="form-control"
              maxlength="150"
              placeholder="Informe a email"
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="col-md-1"
        >
          <button
            class="btn btn-success"
            type="button"
          >
            Salvar
          </button>
        </div>
        <div
          class="col-md-1"
        >
          <button
            class="btn btn-danger"
            type="button"
          >
            Cancelar
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": Object {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;
